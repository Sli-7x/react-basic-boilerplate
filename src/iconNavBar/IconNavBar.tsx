import * as React from 'react';
import { Icon } from '../index';
import { Container, Item } from './IconNavBar.style';

export interface IconNavBarItem {
  id: string;
  title: string;
  icon: string;
}

export interface IKeys {
  id: string;
  title: string;
  icon: string;
}

export interface IIconNavBarProps {
  list: IconNavBarItem[];
  value: string | number;
  name: string;
  keys?: IKeys;
  onSelect: (id: string, name: string) => void;
}

export class IconNavBar extends React.Component<IIconNavBarProps, any> {
  constructor(props: IIconNavBarProps) {
    super(props);
    this.onClick = this.onClick.bind(this);

    this.state = {
      keys: props.keys ? props.keys : { id: 'id', title: 'title', childs: 'childs' },
    };
  }

  onClick(id: string) {
    this.props.onSelect(id, this.props.name);
  }

  render() {
    const { keys } = this.state;
    const { value } = this.props;

    return (
      <Container>
        {this.props.list.map((val: any, i: number) => (
          <Item key={i} selected={value === val[keys.id]}>
            <Icon onClick={this.onClick.bind(this, val[keys.id])} name={val[keys.icon]} />
          </Item>
        ))}
      </Container>
    );
  }
}
