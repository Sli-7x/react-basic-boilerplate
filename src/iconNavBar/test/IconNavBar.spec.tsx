import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { IIconNavBarProps, IconNavBar } from '../IconNavBar';

describe('IconNavBar', () => {
  const onSelect = jest.fn();
  const props: IIconNavBarProps = {
    onSelect,
    value: '',
    name: '',
    list: [
      {
        id: '1',
        title: 'test',
        icon: 'random',
      },
    ],
  };

  it('should render correctly', () => {
    const tree = renderer.create(<IconNavBar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should cover onClick method', () => {
    const tree = shallow(<IconNavBar {...props} />);
    const instance = tree.instance() as IconNavBar;
    instance.onClick('test');
    expect(onSelect).toHaveBeenCalled();
  });

  it('should cover when onSelect() prop not passed', () => {
    const newProps = { ...props };
    delete newProps.onSelect;
    const tree = renderer.create(<IconNavBar {...newProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should cover when prop.keys passed', () => {
    const newProps = { ...props };
    newProps.keys = { id: 'id', title: 'title', icon: 'icon' };
    const tree = renderer.create(<IconNavBar {...newProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
