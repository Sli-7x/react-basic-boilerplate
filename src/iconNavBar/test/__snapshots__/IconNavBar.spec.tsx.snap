// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconNavBar should cover when onSelect() prop not passed 1`] = `
<div
  className="sc-EHOje jTDsAy"
>
  <div
    className="sc-bZQynM icdWAQ"
    selected={false}
  >
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "25px",
        }
      }
      title=""
    >
      
    </i>
  </div>
</div>
`;

exports[`IconNavBar should cover when prop.keys passed 1`] = `
<div
  className="sc-EHOje jTDsAy"
>
  <div
    className="sc-bZQynM icdWAQ"
    selected={false}
  >
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "25px",
        }
      }
      title=""
    >
      random
    </i>
  </div>
</div>
`;

exports[`IconNavBar should render correctly 1`] = `
<div
  className="sc-EHOje jTDsAy"
>
  <div
    className="sc-bZQynM icdWAQ"
    selected={false}
  >
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "25px",
        }
      }
      title=""
    >
      
    </i>
  </div>
</div>
`;
